horizomer/utils/tree.py

```diff
@@ -44,6 +44,31 @@ def compare_topology(tree1, tree2):
     return n2p1 == n2p2
 
 
+def intersect_trees(tree1, tree2):
```

Comment:
high level question? Why do you need to do this? If it is comparing via RF or subsets in skbio, this is done implicitly. You might also want to check skbio's codebase if you cannot "steal" this functionality from there.

Reply:
I can't find if there is an existing scikit-bio function to do this. Do you know?

Reply:
I guess it is this one https://github.com/biocore/scikit-bio/blob/0a9a9f013fe445860b741ac949eceac56b2a6fbb/skbio/tree/_tree.py#L2725

Reply:
Wow, it looks that my function was trying to replicate the `compare_tip_distances` function. However that function does not return two trees. So I have to write one anyway.